transfers/amclient.py

```diff
@@ -0,0 +1,487 @@
+#!/usr/bin/env python
+"""Archivematica Client.
+
+Module and CLI that holds functionality for interacting with the various
+Archivematica APIs.
+
+KNOWN BUG: if you call ./amclient.py with no args/options, you'll get an error.
+See https://github.com/aliles/begins/pull/60 for the PR that fixes this on Py3.
+"""
+from __future__ import print_function, unicode_literals
+import binascii
+import base64
+from collections import defaultdict
+import json
+import logging
+import logging.config  # Has to be imported separately
+import os
+try:
+    from os import fsencode
+except ImportError:
+    def fsencode(filename):
+        """Cribbed & modified from Python3's OS module to support Python2."""
+        encoding = sys.getfilesystemencoding()
+        if isinstance(filename, str):
+            return filename
+        elif isinstance(filename, unicode):
+            return filename.encode(encoding)
+        else:
+            raise TypeError("expect bytes or str, not %s" %
+                            type(filename).__name__)
+import pprint
+import re
+import sys
+
+import begin
+import requests
+from six.moves import configparser
+
+
+THIS_DIR = os.path.abspath(os.path.dirname(__file__))
+DEFAULT_LOGFILE = os.path.join(THIS_DIR, 'amclient.log')
+LOGGER = logging.getLogger('amclient')
+CONFIG_FILE = None
+RETRY_COUNT = 5
+DEF_AM_URL = 'http://127.0.0.1'
+DEF_SS_URL = 'http://127.0.0.1:8000'
+DEF_USER_NAME = 'test'
+UUID_PATT = re.compile(
+    '^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$')
+
+
+def get_setting(setting, default=None):
+    """Get ``setting`` from our config file; return ``default`` if not
+    possible.
+    """
+    config = configparser.SafeConfigParser()
+    try:
+        config.read(CONFIG_FILE)
+        return config.get('amclient', setting)
+    except (TypeError, configparser.Error):
+        return default
+
+
+def _call_url_json(url, params, method='GET'):
+    """Helper to GET a URL where the expected response is 200 with JSON.
+    :param str url: URL to call
+    :param dict params: Params to pass to requests.get
+    :returns: Dict of the returned JSON or None
+    """
+    if method.upper() in ('GET', 'POST', 'DELETE', 'PUT', 'HEAD', 'OPTIONS'):
+        method = method.upper()
+    else:
+        method = 'GET'
```

Comment:
Could this if/else be `method = method.upper()`?  That would allow eg PATCH, requests already defaults to GET, and so is the default parameter of this function.